frontend/index.js

```diff
@@ -1038,6 +1059,28 @@ async function bufferFromStream(stream) {
   return buffer
 }
 
+async function updateHistory() {
+  const hash = await contentHash()
+  const currentHash = window.location.hash
+  if (hash === currentHash) {
+    // no changes detected
+    return
+  }
```

Comment:
As mentioned in PR notes, this isn't the most efficient way to check for code changes - but seemed like a good enough first pass since alternatives would have involved adding additional state variables at the top level (something I wanted to avoid)